src/components/Portal/Portal.spec.tsx

```diff
@@ -15,6 +15,15 @@ describe('Portal', () => {
 			});
 		});
 
+		describe('data-test-id', () => {
+			it('should set the data-test-id of the portal DOM element portalId', () => {
+				const wrapper = shallow(<Portal className='test-classname' />);
+				expect(
+					wrapper.find('[data-test-id="test-classname"]').exists()
```

Comment:
Same as above `data-testid`

Also, I'm a little confused why the className is being passed to the data-testid.  We should just create a new prop name dataTestid and pass that down to the portal.

Reply:
I do use the className and not separate props passing down the component because this change will just add our `data-test-id` to elements without doing additional changes to the parent components calling this.

If we use a separate prop, it will have affects on other parent component calling this and change will be large enough which seems to me an additional effort for this.

Reply:
@swetanshkumar I think it makes more sense to follow @n8thegr8's suggestion.  This is an open source library and I think consumers would expect that you could pass down a data-testid prop that would get added to the component with the same prop name.  I don't think using the classname for the data-testid prop would be expected.

Reply:
@semaurer , then this might more changes to the parent component used. Ideally we don't directly call lucid portals in our apps, its generally the parent components which passed down the props and portal is being rendered.

For example - 
Take Dialog as an example, We generally use that pretty much. So the flow is - 

Dialog ---> Overlay ---> Portals

if we introduce a separate prop, we might have to pass it from Dialog as well. So its like you have to update Dialog to accept data-test-id as a prop to pass down to Overlay, then add data-test-id to Overlay as a prop to accept and pass it along Portals


Note - This is just an example of Dialog. There are still many (SearchableMultiSelect ---> DropMenu ---> ContextMenu ---> Portals)

Reply:
data-test-ids should be unique whereas class names are not.

Reply:
We currently do not use `data-test-id` props much at all within Lucid — you'll find them only on the "Skeleton" family of components at the moment, and they are certainly not unique — so there is an established precedent within the library.

I agree with @swetanshkumar that @n8thegr8's suggestion is a bigger discussion that we need to have insofar as how we're going to handle this prop across the board in a consistent manner for all of the published components. If we cover that in this particular PR, then as @swetanshkumar said we would need to update not just the Portal component but every component within Lucid that consumes that component.

An interim solution would be to set a default value for the `data-test-id` prop that utilizes some sort of unique ID generator in order to ensure uniqueness, but I strongly feel that we should look to handling this uniformly across the component library through a deliberate, concerted effort rather than a temporary, one-off solution.